tests/unit/test_configuration.py

```diff
@@ -51,26 +50,26 @@ def test_default_configuration_structure() -> None:
     """Test that the default configuration structure is correct"""
     config = get_configuration()
     # Check datasets
-    assert set(f.name for f in DATASET_SCHEMAS) == set(config.keys())
+    assert set(f.name for f in DATASET_SCHEMAS) == set(config.to_dict().keys())
     for dataset_schema in DATASET_SCHEMAS:
         # Check row noise
         for row_noise in dataset_schema.row_noise_types:
-            config_probability: LayeredConfigTree = config[dataset_schema.name][
-                Keys.ROW_NOISE
-            ][row_noise.name]
+            config_probability: dict = config.to_dict()[dataset_schema.name][Keys.ROW_NOISE][
+                row_noise.name
+            ]
             validate_noise_type_config(dataset_schema, row_noise, config_probability)
         for col in dataset_schema.columns:
             for noise_type in col.noise_types:
-                config_level: LayeredConfigTree = config[dataset_schema.name][
-                    Keys.COLUMN_NOISE
-                ][col.name][noise_type.name]
+                config_level: dict = config.to_dict()[dataset_schema.name][Keys.COLUMN_NOISE][
+                    col.name
+                ][noise_type.name]
                 validate_noise_type_config(dataset_schema, noise_type, config_level, col)
 
 
 def validate_noise_type_config(
     dataset_schema: DatasetSchema,
     noise_type: NoiseType,
-    config_level: LayeredConfigTree,
+    config_level: dict,
```

Comment:
You should be passing a `NoiseConfiguration` object through here, not a dictionary